obs-outputs: Add metadata info for enhanced RTMP multivideotracks

murillo128 opened the pull request:
As specified in latest version of the enhanced RTMP spec:
https://github.com/veovera/enhanced-rtmp/blob/main/docs/enhanced/enhanced-rtmp-v2.md#enhancing-onmetadata

### Motivation and Context
As discussed in https://github.com/veovera/enhanced-rtmp/issues/37
there is currently no way of retrieving the metadata of the enhanced streaming multivideotrack which are not the main one.

The latest of the version has included an extension to the `onMetaData` object to include that info, and this PR implements it.

### How Has This Been Tested?
It has been tested against the millicast rtmp server supporting enhanced streaming

### Types of changes
Tweak (non-breaking change to improve existing functionality) 

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

derrod commented:
Since the spec has not actually been updated to include these changes (only the draft on a different branch), this PR should also be marked as a draft for now.

murillo128 commented:
@derrod fair enough, could it be reviewed even in draft so we can merge it as soon as the new specification is available?

murillo128 commented:
The PR has been merged into the main branch of the spec.

murillo128 commented:
@derrod ping?

murillo128 commented:
the enhanced rtmp containing the change has gone into beta v2-2024-10-22-b1 https://github.com/veovera/enhanced-rtmp/commit/82bfd58e06e5329969bda73d62521aa298b7a18b

murillo128 commented:
@derrod the ertmp v2 spec has been finally released

derrod commented on plugins/obs-outputs/flv-mux.c:
```diff
@@ -228,6 +238,41 @@ static void build_flv_meta_data(obs_output_t *context, uint8_t **output, size_t
+				enc_num_val(&enc, end, "width",
+					    (double)obs_encoder_get_width(
+						    vencoder_multi));
+				enc_num_val(&enc, end, "height",
```
I'm unsure why CI doesn't complain, but you should run this through `clang-format` to adjust to the new 120 column line limit.

derrod reviewed (approved):
Seems fine other than formatting.

murillo128 committed:
obs-outputs: Add metadata info for enhanced RTMP multivideotracks

As specified in latest version of the enhanced RTMP spec:
https://github.com/veovera/enhanced-rtmp/blob/main/docs/enhanced/enhanced-rtmp-v2.md#enhancing-onmetadata

murillo128 committed:
Merge branch 'master' into multivideotrack-metadata

murillo128 commented:
any update? could this be merged?
